pandas/io/parsers/base_parser.py

```diff
@@ -114,6 +115,8 @@
 class ParserBase:
     _implicit_index: bool = False
     _first_chunk: bool
+    index_col: int | Sequence[int] | None
```

Comment:
index_col should only be a Sequence here because of
https://github.com/pandas-dev/pandas/blob/1a20d13079f05aaa75f8a1153d2a18dd0b9931ee/pandas/io/parsers/readers.py#L956-L958
index_col might also be a Sequence[str] here too.

Reply:
so the annotation on 118 should just be `index_col: Sequence[int] | Sequence[str]`?

Reply:
Looking again, it actually might be ``index_col: Sequence[int] | Sequence[str] | Literal[False] | None``.

Reply:
Looking at this i think a) you're right and b) this is an unholy mess; we should do e.g. the int -> list[int] check/conversion just once up-front